praw/models/reddit/comment.py

```diff
@@ -13,6 +13,16 @@ class Comment(RedditBase, InboxableMixin, UserContentMixin):
                                'comment tree')
     STR_FIELD = 'id'
 
+    @staticmethod
+    def id_from_url(url):
+        """Get the ID of a comment from the full URL."""
+        try:
+            comment_id = url.rsplit('://')[-1].rsplit('/')[6].rsplit('?')[0]
+        except IndexError:  # Link is shorter than normal permalink
```

Comment:
Can you use urlparse to do the initial parsing, and then use `rsplit` just on the path portion? In that way we shouldn't need exception handling.